openoa/utils/plot.py

```diff
@@ -1932,3 +1932,284 @@ def plot_wake_losses(
+            misalignment plots are returned for further tinkering/saving.
+    """
+
+    from openoa.analysis.yaw_misalignment import cos_curve
+
+    power_color_code = "#4477AA"
+    mean_vane_color_code = "#000000"
+    curve_fit_color_code = "#EE6677"
+
+    plot_kwargs_fill.setdefault("alpha", 0.2)
+
+    if figure_kwargs is None:
+        figure_kwargs = {}
+
+    # determine if confidence intervals should be plotted (i.e., UQ) based on dimension of data
+    if (
+        (power_values_vane_ws.ndim == 2)
+        & (curve_fit_params_ws.ndim == 2)
+        & (yaw_misalignment_ws.ndim == 1)
+    ):
+        UQ = False
+    elif (
+        (power_values_vane_ws.ndim == 3)
+        & (curve_fit_params_ws.ndim == 3)
+        & (yaw_misalignment_ws.ndim == 2)
+    ):
+        UQ = True
+    else:
+        raise ValueError(
+            "The inputs `power_values_vane_ws`, `curve_fit_params_ws`, and `yaw_misalignment_ws` "
+            "must have either 2, 2, and 1 dimensions or 3, 3, and 2 dimensions, respectively."
+        )
+
+    # Select figure size based on number of subplots
+    if len(ws_bins) == 1:
+        figure_kwargs.setdefault("figsize", (6, 5))
+        fig, axs = plt.subplots(1, 1, **figure_kwargs)
+        axs = [[axs]]
+        N_col = 1
+    elif len(ws_bins) == 2:
+        figure_kwargs.setdefault("figsize", (11, 5))
+        fig, axs = plt.subplots(1, 2, sharex=True, **figure_kwargs)
+        N_col = 2
+    elif len(ws_bins) == 3:
+        figure_kwargs.setdefault("figsize", (16, 5))
+        fig, axs = plt.subplots(1, 3, sharex=True, **figure_kwargs)
+        N_col = 3
+    elif len(ws_bins) == 4:
+        figure_kwargs.setdefault("figsize", (11, 10))
+        fig, axs = plt.subplots(2, 2, sharex=True, **figure_kwargs)
+        N_col = 2
+    elif len(ws_bins) >= 13:
+        figure_kwargs.setdefault("figsize", (16, 17))
+        fig, axs = plt.subplots(
+            int(np.floor((len(ws_bins) - 1) / 3) + 1), 3, sharex=True, **figure_kwargs
+        )
+        N_col = 3
+    elif len(ws_bins) >= 10:
+        figure_kwargs.setdefault("figsize", (16, 14))
+        fig, axs = plt.subplots(4, 3, sharex=True, **figure_kwargs)
+        N_col = 3
+    elif len(ws_bins) >= 7:
+        figure_kwargs.setdefault("figsize", (16, 11))
+        fig, axs = plt.subplots(3, 3, sharex=True, **figure_kwargs)
+        N_col = 3
+    elif len(ws_bins) >= 5:
+        figure_kwargs.setdefault("figsize", (16, 7.5))
+        fig, axs = plt.subplots(2, 3, sharex=True, **figure_kwargs)
+        N_col = 3
+
+    for i, ws in enumerate(ws_bins):
+        if UQ:
+            norm_factor = curve_fit_params_ws[:, i, 0].mean()
+
+            y_min = np.nanmin(np.nanpercentile(power_values_vane_ws[:, i, :], 2.5, 0))
+            y_max = np.nanmax(np.nanpercentile(power_values_vane_ws[:, i, :], 97.5, 0))
+
+            axs[int(np.floor(i / N_col))][i % N_col].fill_between(
```

Comment:
It seems like it'd be best if the axis in `axs` could be pulled at the top of the for loop to reduce the number of times this index is calculated to help make it easier to read. Typically, I prefer something like `ax = axs[int(np.floor(i / N_col))][i % N_col]`, so each subsequent call can just be `ax.plot()`.

Reply:
Good suggestion. This has been updated.